src/main/java/core/basesyntax/dao/StorageDaoImpl.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax.dao;
+
+import core.basesyntax.storage.Storage;
+import java.util.Map;
+
+public class StorageDaoImpl implements StorageDao {
+    private Storage storage;
+
+    public StorageDaoImpl() {
+        storage = new Storage();
+    }
+
+    @Override
+    public void save(String fruit, int quantity) {
+        if (fruit == null) {
+            throw new RuntimeException("fruit cannot be null");
+        }
+        storage.getFruitStorage().put(fruit, quantity);
+    }
+
+    @Override
+    public Map<String, Integer> getAll() {
+        return storage.getFruitStorage();
+    }
+
+    @Override
+    public int getQuantityByFruitName(String key) {
+        return storage.getFruitStorage().get(key);
```

Comment:
The method `getQuantityByFruitName` may throw a `NullPointerException` if the fruit name does not exist in the storage. Consider using `getOrDefault` or checking if the key exists before accessing it.